sources/api/corndog/src/main.rs

```diff
@@ -45,20 +50,29 @@ fn run() -> Result<()> {
     SimpleLogger::init(args.log_level, LogConfig::default()).context(error::LoggerSnafu)?;
 
     // If the user has kernel settings, apply them.
-    let kernel = get_kernel_settings(args.config_path)?;
     match args.subcommand.as_ref() {
         "sysctl" => {
+            let kernel = get_kernel_settings(args.config_path)?;
             if let Some(sysctls) = kernel.sysctl {
                 debug!("Applying sysctls: {:#?}", sysctls);
                 set_sysctls(sysctls);
             }
         }
         "lockdown" => {
+            let kernel = get_kernel_settings(args.config_path)?;
             if let Some(lockdown) = kernel.lockdown {
                 debug!("Setting lockdown: {:#?}", lockdown);
                 set_lockdown(&lockdown)?;
             }
         }
+        "generate-hugepages-setting" => {
+            let mut hugepages: String = "0".to_string();
```

Comment:
That is fair. Will update that.

Reply:
Updated, wanted to hear your thoughts on my implementation. The implementation would try to read the `nr_hugepages` and try to parse it. It would set to "0" if it cannot recognize the value read from the file.

The hugepages is expected to be an unsigned long value, see https://github.com/torvalds/linux/blob/master/include/linux/hugetlb.h#L673.